backend/app/deployment_handler.go

```diff
@@ -722,3 +775,248 @@ func (h *Handler) HandleRemoveNode(c *gin.Context) {
 		Message:    "Node removal workflow started successfully",
 	})
 }
+
+// @Summary Deploy a VM
+// @Description Creates and deploy a virtual machine
+// @Tags vms
+// @Security BearerAuth
+// @Accept json
+// @Produce json
+// @Param vm body DeployVMInput true "VM configuration"
+// @Success 202 {object} DeployVMResponse "WorkflowID and Status"
+// @Failure 400 {object} APIResponse "Invalid request"
+// @Failure 401 {object} APIResponse "Unauthorized"
+// @Failure 500 {object} APIResponse "Internal server error"
+// @Router /deployments/vms [post]
+func (h *Handler) HandleDeployVM(c *gin.Context) {
+	config, err := h.getClientConfig(c)
+	if err != nil {
+		InternalServerError(c)
+		return
+	}
+	var input DeployVMInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		Error(c, http.StatusBadRequest, "Invalid request json format", err.Error())
+		return
+	}
+
+	if err := internal.ValidateStruct(input.Node); err != nil {
+		Error(c, http.StatusBadRequest, "Validation failed", err.Error())
+		return
+	}
+
+	// Map to internal kubedeployer.VM
+	vm := kubedeployer.VM{
```

Comment:
this is no longer a kubedeployer, I believe there're shared bits that kubedeployer and vmdeployer can use, but not like this.